### PR TITLE
Replace hand-rolled HTML sanitizer with DOMPurify and add link styling

### PR DESCRIPTION
**Pull request by pstaylor-patrick**

### CODE REVIEW

**Review by pstaylor-patrick — COMMENTED** [review]
## Code Review Summary

This PR replaces the `stripHtml` SSR fallback with a regex-based `serverSanitize()` that preserves anchor tags as clickable React links during server-side rendering. Good improvement that eliminates hydration mismatch and keeps links functional in SSR.

**What works well:**
- Clean separation of server vs client sanitization paths
- Proper `isSafeHref` validation blocks `javascript:` protocol
- `target="_blank"` with `rel="noopener noreferrer"` on all links
- Good test coverage for the server-side path (10 tests)
- Consistent link styling via shared `LINK_STYLE` constant

**Top risks:**
- The inherited script/style stripping regex can fail on content containing `<` characters (e.g., comparison operators in script blocks) -- this was pre-existing but now is the sole SSR defense
- Regex-based HTML parsing has inherent limitations with malformed/nested markup

## Findings

| ID | Severity | Title |
|----|----------|-------|
| P1 | warning | Regex anchor match can mismatch on nested/malformed anchors |
| P2 | warning | Href extraction skips unquoted href attributes |
| P3 | warning | Script/style stripping regex bypassed by `<` in content |
| P4 | suggestion | Duplicate link-rendering logic between server and client paths |
| P5 | suggestion | Missing test for script tags with `<` in content |

No blockers found. Solid PR overall -- the warnings are edge cases worth being aware of but not merge-blocking given the controlled data source.

File: src/utils/safeHtml.tsx [hunk 1]
```diff
@@ -1,15 +1,76 @@
-import type { ReactNode } from 'react';
+import type { CSSProperties, ReactNode } from 'react';
 
 const ALLOWED_LINK_PROTOCOLS = new Set(['http:', 'https:', 'mailto:', 'tel:']);
+const LINK_STYLE: CSSProperties = {
```
[the remark this answers is not in the document]
Comment (pstaylor-patrick) [reply src/utils/safeHtml.tsx]: Resolved: The rewrite to DOMPurify eliminated the duplicate rendering logic. There is now a single `sanitizeHtml()` function with one code path.

**Review by pstaylor-patrick — COMMENTED** [review]
## Re-Review Summary

The latest commit (`42d5ead`) replaced the hand-rolled regex sanitizer with `isomorphic-dompurify`. This is a major improvement that resolves all 5 findings from the previous review (P1-P5). DOMPurify is a well-maintained, battle-tested library for HTML sanitization.

All previous findings have been marked resolved. However, the rewrite introduces one new blocker and one new warning:

- **P6 [blocker]** Fallback to unsanitized HTML in `dangerouslySetInnerHTML`
- **P7 [warning]** Major version mismatch: `jest@29` with `jest-environment-jsdom@30`

**Review by pstaylor-patrick — COMMENTED** [review]
## Re-review: CLEAN (would APPROVE, but GitHub blocks self-approval)

All 7 findings from prior rounds (P1-P7) have been resolved. Fresh review of the complete changeset found no remaining issues.

### Summary

- **Security**: DOMPurify (via isomorphic-dompurify) is a well-maintained, battle-tested sanitizer. The allowlisted tags/attributes are appropriately restrictive. `ALLOWED_URI_REGEXP` blocks javascript:/data: URIs. The `afterSanitizeAttributes` hook correctly forces `target="_blank"` and `rel="noopener noreferrer"` on all anchors.
- **Correctness**: `enrichEmailsInHtml` runs *before* `sanitizeHtml` in WorkoutNotes, which is the right order -- raw email addresses get wrapped in `<a>` tags, then DOMPurify validates those mailto links and strips anything unsafe.
- **XSS surface**: The `dangerouslySetInnerHTML` usage in RegionContent and WorkoutNotes is safe because all inputs pass through DOMPurify first. The previous `|| descriptionSource` fallback (P6) has been removed.
- **Dependencies**: `jest-environment-jsdom` is pinned at `^29.7.0` matching the `jest` major version (P7 fix confirmed).
- **Testing**: 13 test cases cover sanitization (script/style stripping, javascript: URI blocking, event handler stripping, data: URI blocking, img stripping) and email enrichment.
- **DOMPurify hook is module-scoped**: The `addHook` call runs once at module load on the global DOMPurify singleton. This is fine for this codebase since there is a single sanitization policy, but worth noting if a second policy is ever needed.

No blockers, warnings, or suggestions. Ship it.

**Comment by pstaylor-patrick** [comment]
Closing as part of an audit of my authored PRs/issues to clear out stale items (no activity in 30+ days). Not a judgment on the underlying idea — please reopen if this is still a priority.
